File: sdk/js/freewayai/src/database_query_template_format.ts
import { _format_template_string, _read_query_from_system_and_id } from './utilities';

interface Query {
  search_query_template: string;
}

function _format_qery(query: Query, variables: Record<string, any>): string {
  const queryTemplate = query.search_query_template;
  return _format_template_string(query, queryTemplate, variables);
}

function read_and_format_query(systemId: string, systemsLocation: string, queryId: string, variables: Record<string, any>): string {
  const query = _read_query_from_system_and_id(systemId, systemsLocation, queryId);
  return _format_qery(query, variables);
}

export { read_and_format_query };

File: sdk/python/src/freewayai/database_query_template_format.py
from .utilities import format_template_string, read_query_from_system_and_id

# FreewayAI is property of DiligentlyAI, Inc. and is used under an MIT license.
# by Ben Parfitt, Feb 2024

def _format_query(query, variables):
    query_template = query["search_query_template"]
    return format_template_string(query, query_template, variables)
    

def read_and_format_query(system_id, systems_location, query_id, variables):
    prompt = read_query_from_system_and_id(system_id, systems_location, query_id)
    return _format_query(prompt, variables)